src/platform/imx8m/lib/dai.c

```diff
@@ -34,6 +39,28 @@ static SHARED_DATA struct dai sai[] = {
 	},
 	.drv = &sai_driver,
 },
+
+{
+	.index = 2,
+	.plat_data = {
+		.base = SAI_2_BASE,
+		.fifo[SOF_IPC_STREAM_PLAYBACK] = {
+			.offset = SAI_2_BASE + REG_SAI_TDR0,
+			.depth = 128, /* in 4 bytes words */
+			.watermark = 64, /* half the depth */
+			/* Handshake is SDMA hardware event */
+			.handshake = 3,
```

Comment:
Did you verify these handshake numbers? You can find the SDMA hardware event numbers in the i.MX8MP reference manual. The naming "handshake" is an unfortunate consequence on how SOF defined the structure in the first place, I have used the "handshake" to provide information specific to the DMA driver (in the case of SDMA, the hardware event which is to be used)